### PR TITLE
Conda recipe for nbpresent

### PR DESCRIPTION
**Pull request by ijstokes**
This provides a conda recipe for nbpresent.

### CODE REVIEW

File: conda.recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,11 @@
+npm install .
+npm run build
+"%PYTHON%" setup.py install
+"%PYTHON%" -m nbpresent.install --force --enable
+if errorlevel 1 exit 1
+
+:: Add more build steps here, if they are necessary.
```
Comment (damianavila) [inline conda.recipe/bld.bat]: Are we sure we want those comment below? I don't think so...

File: conda.recipe/meta.yaml [hunk 2]
```diff
@@ -5,14 +5,15 @@ package:
 extra:
   channels:
     - bokeh
+    - cpcloud
```
Comment (damianavila) [inline conda.recipe/meta.yaml]: why do you need cpcloud here?

File: conda.recipe/meta.yaml [hunk 3]
```diff
@@ -5,14 +5,15 @@ package:
 extra:
   channels:
     - bokeh
```
Comment (damianavila) [inline conda.recipe/meta.yaml]: We should use the javascript channel here...

File: conda.recipe/meta.yaml [hunk 4]
```diff
@@ -5,14 +5,15 @@ package:
...
+    - cpcloud
 
-build:
-  number: 1
-  script:
-    - ls
-    - npm install .
-    - npm run build
-    - python setup.py install
+#build:
```
Comment (damianavila) [inline conda.recipe/meta.yaml]: Commented code should be deleted...

File: conda.recipe/meta.yaml [hunk 5]
```diff
@@ -20,7 +21,9 @@ source:
 requirements:
   build:
     - nodejs
+    - npm
```
Comment (damianavila) [inline conda.recipe/meta.yaml]: `npm` comes for free when you install `nodejs` (this is why you are adding cpcloud channel?), so no need to add it...

File: conda.recipe/build.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+SRC_DIR=$RECIPE_DIR/..
+cd $SRC_DIR
+npm install .
+npm run build
+python setup.py install
+python -m nbpresent.install --force --enable
```
Comment (damianavila) [inline conda.recipe/build.sh]: Technically, this line should be put in a post-script file: http://conda.pydata.org/docs/building/build-scripts.html

File: conda.recipe/bld.bat [hunk 7]
```diff
@@ -0,0 +1,11 @@
+npm install .
+npm run build
+"%PYTHON%" setup.py install
+"%PYTHON%" -m nbpresent.install --force --enable
```
Comment (damianavila) [inline conda.recipe/bld.bat]: Technically, this line should be put in a post-script file: http://conda.pydata.org/docs/building/build-scripts.html

**Comment by damianavila** [comment]
@ijstokes @bollwyvl, I leave some comments on the PR :wink:

**Comment by bollwyvl** [comment]
This is looking great. I'll review presently.

First snag: I am initially hitting `Error: No packages found in current linux-64 channels matching: npm` here on my Fedora 22 box. Will poke around some.

File: conda.recipe/meta.yaml (hunk 5 above)
Comment (bollwyvl) [inline conda.recipe/meta.yaml]: I can confirm this...

File: conda.recipe/build.sh (hunk 6 above)
Comment (bollwyvl) [inline conda.recipe/build.sh]: Thanks for the link!

The install `nbpresent.install` actually needs to be run on install.

I also had to add `--user` to allow it to work on my linux machine. @damianavila in your experience, where do most people install extensions (i.e. global/user)? Is there an expectation that the global `/share` location is really shared by every user/app?

Comment (damianavila) [inline conda.recipe/build.sh]: >@damianavila in your experience, where do most people install extensions (i.e. global/user)? 

Usually in user space...

>Is there an expectation that the global /share location is really shared by every user/app?

In some deployment like JupyterHub, maybe it makes sense to have a system-wide installation...

Comment (bollwyvl) [inline conda.recipe/build.sh]: Is there a handy way to do runtime flags? Checking the doc...

Comment (bollwyvl) [inline conda.recipe/build.sh]: Sorry, install-time. I'm thinking `[+user]` vs `[+global]`.

**Comment by bollwyvl** [comment]
:shipit:

File: conda.recipe/build.sh (hunk 6 above)
Comment (damianavila) [inline conda.recipe/build.sh]: > Sorry, install-time. I'm thinking `[+user]` vs `[+global]`.

Not sure there are runtime env variables available for conda... if I have to bet, I would say it does not exist... but maybe you surprise me :wink:

**Pull request merged**